### PR TITLE
update bowtie2 wrapper

### PR DESCRIPTION
**Pull request by bluegenes**
enable multiple files per sample

### CODE REVIEW

**Comment by bluegenes** [comment]
Failing on tests unrelated to this PR

```
test.py::test_bowtie2_align PASSED                                       [  2%]
test.py::test_bowtie2_index PASSED                                       [  3%]
```

**Comment by bluegenes** [comment]
ready for review

File: bio/bowtie2/align/test/Snakefile [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
         extra=""  # optional parameters
-    threads: 8
+    threads: 4
+    wrapper:
+        "master/bio/bowtie2/align"
+
+rule bowtie2_single:
+    input:
+        single=["reads/{sample}.1.fastq", "reads/{sample}.2.fastq"],
+        index_donefile="index/genome.bowtie2index.done" # required for workflows to ensure index is built first
```
Comment (johanneskoester) [inline bio/bowtie2/align/test/Snakefile]: I prefer to not impose such handling in the wrapper examples for now. The better solution would be to have the index files as true input files. But it depends on the workflow whether that is needed, or rather a directory should be passed or even nothing. Hence, let us not add this to the wrapper.

Comment (bluegenes) [reply bio/bowtie2/align/test/Snakefile]: I had an option written for an index file in the inputs, but I can roll back if you think it's better as a param

File: bio/bowtie2/align/wrapper.py [hunk 2]
```diff
@@ -1,26 +1,65 @@
...
+index = snakemake.input.get("index")
+try:
+    index_dirname = os.path.dirname(index)
+    # get index prefix from index file. pattern should be <prefix>.(\d).bt2 or <prefix>.rev.(\d).bt2
+    index_prefix = re.match("^([^.]*).(rev.)?\d+.bt2", os.path.basename(index)).groups()[0]
+    index_prefix = os.path.join(index_dirname, index_prefix)
+except:
+    raise ValueError(
+        "please index your assembly with bowtie2-build and provide an index file (e.g. with '.1.bt2' extension) via the 'index' input param"
+    )
```
Comment (bluegenes) [inline bio/bowtie2/align/wrapper.py]: @johanneskoester modified index section

**Comment by johanneskoester** [comment]
The next release of snakemake will fix the index situation. Let us postpone the merge until then, hopefully very soon.

**Comment by johanneskoester** [comment]
Ok, the index can now be written as input file and defined via [multiext](https://snakemake.readthedocs.io/en/stable/snakefiles/rules.html#the-multiext-function). Can you update the PR accordingly?

**Comment by dlaehnemann** [comment]
@bluegenes Any time for a quick update of this PR to use `multiext()` for the index and resolve the conflicts? Or should fork your branch and start a new PR?

**Comment by johanneskoester** [comment]
Closing because of inactivity. I think it would be best to start over using multiext, similar to the bwa-mem wrapper changes that were recently merged.

**Pull request closed without merging**